### PR TITLE
Added getBundleUrl static method to lazy components

### PR DESCRIPTION
**Pull request by rohan-deshpande**
### Added 

* `getBundleUrl` static method to lazy components + tests for this
* `preload` static method to lazy components (not working yet)

### Removed 

* `prefetch` and `Prefetch` static methods for lazy components

### CODE REVIEW

File: src/__tests__/unit.test.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { lazyForPaint } from '..';
```
Comment (MonicaOlejniczak) [inline src/__tests__/unit.test.tsx:1]: So generally unit tests should live closer to the actual unit, and top level tests folder are either reserved for integration style tests, or files that are in the root directory. Could this file be moved to the `lazy` directory under its own `__tests__` folder, or introduce a `.test.tsx` pattern and place it appropriately?

File: src/__tests__/unit.test.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const mockModuleId = '@foo/bar';
+    const lazyComponent = lazyForPaint(
+      // @ts-ignore - mocking import()
+      () => Promise.resolve({ default: mockModuleId }),
+      {
+        getCacheId: () => '',
+        moduleId: mockModuleId,
+      }
+    );
+    describe('getBundleUrl', () => {
```
Comment (MonicaOlejniczak) [inline src/__tests__/unit.test.tsx:21]: /nit Line break above

File: src/__tests__/unit.test.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      it('should return a LazyComponent with a getBundleUrl method on it', () => {
```
Comment (MonicaOlejniczak) [inline src/__tests__/unit.test.tsx:22]: Is this test necessary? The below test also captures this, and should be enforced by the type system if possible

File: src/__tests__/unit.test.tsx (hunk 1 above)
Comment (rohan-deshpande) [reply src/__tests__/unit.test.tsx:1]: I was just following the convention established here. Happy to move it if that's what we want to do

File: src/__tests__/unit.test.tsx (hunk 2 above)
Comment (rohan-deshpande) [reply src/__tests__/unit.test.tsx:21]: Prettier says no

File: src/__tests__/unit.test.tsx (hunk 3 above)
Comment (rohan-deshpande) [reply src/__tests__/unit.test.tsx:22]: Probably not, will remove

File: src/__tests__/unit.test.tsx (hunk 2 above)
Comment (rohan-deshpande) [reply src/__tests__/unit.test.tsx]: Done

**Pull request merged**